app/assets/javascripts/discourse/tests/integration/components/float-kit/d-menu-test.js

```diff
@@ -350,4 +350,43 @@ module("Integration | Component | FloatKit | d-menu", function (hooks) {
     assert.dom(".fk-d-menu__trigger.first").doesNotExist();
     assert.dom(".fk-d-menu.first").exists();
   });
+
+  test("focusTrigger on close", async function (assert) {
+    this.api = null;
+    this.onRegisterApi = (api) => (this.api = api);
+    this.close = async () => await this.api.close();
+
+    await render(
+      hbs`<DMenu @onRegisterApi={{this.onRegisterApi}} @inline={{true}} @icon="xmark">
+        <DButton @icon="xmark" class="close" @action={{this.close}} />
+      </DMenu>`
+    );
+
+    await click(".fk-d-menu__trigger");
+    await triggerKeyEvent(document.activeElement, "keydown", "Tab");
+    await triggerKeyEvent(document.activeElement, "keydown", "Enter");
+
+    assert.strictEqual(
+      document.activeElement,
+      document.querySelector(".fk-d-menu__trigger")
+    );
```

Comment:
we've got `assert.dom("selector").isFocused()` for that!

Reply:
ah thanks didnt know this one